nh_clinical/wizard/user_allocation.py

```diff
@@ -458,14 +458,28 @@ def deallocate(self, cr, uid, ids, context=None):
         if not isinstance(ids, list):
             ids = [ids]
         wiz = self.browse(cr, uid, ids[0], context=context)
+        deallocate_location_ids = \
+            [location.id for location in wiz.location_ids]
+
         user_pool = self.pool['res.users']
-        location_ids = [location.id for location in wiz.location_ids]
-        user_ids = user_pool.search(
+        all_doctor_ids = user_pool.search(
             cr, uid, [['groups_id.name', 'in', self._doctor_groups]],
-            context=context)
-        user_pool.write(cr, uid, user_ids,
-                        {'location_ids': [[5, location_ids]]},
-                        context=context)
+            context=context
+        )
+
+        for doctor_id in all_doctor_ids:
+            doctor_current_location_ids = user_pool.read(
+                cr, uid, doctor_id, fields=['location_ids']
+            )['location_ids']
+            doctor_new_location_ids = \
```

Comment:
I did actually try that but was surprised to find `-` isn't a valid operator on lists. It works on sets: `doctor_new_location_ids = set(doctor_current_location_ids) - set(deallocate_location_ids)`

It should be fine to use the sets in theory because the list should always be a set anyway but I was taking a paranoid attitude towards it; if there ever are duplicate IDs because of some other code breaking then it will hide that fact. What do you reckon?